src/components/MediatedRequestsActivities/MediatedRequestsActivities.js

```diff
@@ -34,15 +41,117 @@ import {
   MEDIATED_REQUESTS_RECORD_FIELD_NAME,
   MEDIATED_REQUESTS_RECORD_TRANSLATIONS,
   ICONS,
+  SEARCH_FIELDS,
+  FILTER_CONFIG,
+  REPORT_HEADERS,
   getMediatedRequestsActivitiesUrl,
 } from '../../constants';
-import { getTotalCount } from '../../utils';
+import {
+  getTotalCount,
+  modifyRecordsToExport,
+} from '../../utils';
 
-export const getActionMenu = (renderColumnsMenu, history) => () => {
+export const getActionMenu = ({
+  renderColumnsMenu,
+  mediatedRequests,
+  searchValue,
+  activeFilters,
+  reportRecords,
+  callout,
+  history,
+  isLoadingReport,
+  setIsLoadingReport,
+  formatMessage,
+  // eslint-disable-next-line react/prop-types
+}) => ({ onToggle }) => { // NOSONAR
   const goToNewMediatedRequest = () => {
     history.push(`${getMediatedRequestsActivitiesUrl()}/create`);
   };
 
+  const getRecords = async (query) => {
+    const {
+      GET,
+      reset,
+    } = reportRecords;
+
+    const limit = 1000;
```

Comment:
@BogdanDenis 1000 records per request was discussed with BE team and was approved by them.
I can not answer about 100,000 records. Theoretically it could be, but in practice I'm not sure.

Reply:
We use limit param that equals 1000 on `ui-requests` and the same value we use here. And once again it was approved by BE team.

Reply:
@artem-blazhko that's the only concern that I have about this feature - usually csv files are generated on BE side and sent to UI because it could be a large amount of records. But if you discussed the performance aspects of it and you think it won't cause any issues then I have no objections